fastify.js

```diff
@@ -147,7 +148,24 @@ function build (options) {
   fastify.route = route
   fastify._routePrefix = ''
   fastify._logLevel = ''
-
+  if (options.trustProxy) {
+    const tp = options.trustProxy
+    if (typeof tp === 'function') {
+      fastify._trustProxyFn = tp
+    } else if (tp === true) {
+      // Support plain true/false
+      fastify._trustProxyFn = function () { return true }
+    } else if (typeof tp === 'number') {
+      // Support trusting hop count
+      fastify._trustProxyFn = function (a, i) { return i < tp }
+    } else if (typeof tp === 'string') {
+      // Support comma-separated tps
+      const vals = tp.split(',').map(it => it.trim())
+      fastify._trustProxyFn = proxyAddr.compile(vals)
+    } else {
+      fastify._trustProxyFn = proxyAddr.compile(tp || [])
+    }
+  }
```

Comment:
Can you please extract this logic into a function? Also, I would avoid attaching a function to the main fastify object if we can avoid it.